internal/prometheus/prometheus.go

```diff
@@ -21,7 +21,7 @@ import (
 // Provider wraps a metrics.Metrics provider with a Prometheus registry that can
 // instrument the HTTP server's handlers.
 type Provider struct {
-	registry             *prometheus.Registry
+	Registry             *prometheus.Registry
```

Comment:
Can we avoid making the field public? I suppose something like

```go
type ProviderRegisterer interface {
  MustRegister(...)
}
```
(or an error-returning variant) could be defined instead, and used in the plugin code's type assertion.

Reply:
Hey @srenatus , Thanks for this comment,  I’ve just changed it on https://github.com/open-policy-agent/opa/pull/4251/commits/e6d8f8178629bf1174fcc639eb9c226a35d0eb1b. Now I’m passing a prometheus.Registerer interface for the plugin, and it can be used in the future to enable others plugins to register metrics. If you have a better idea or questions  let me know ;)